src/Spice86.Core/Emulator/OperatingSystem/DosPathResolver.cs

```diff
@@ -0,0 +1,235 @@
+namespace Spice86.Core.Emulator.OperatingSystem;
+
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+using Spice86.Core.Emulator.OperatingSystem.Enums;
+using Spice86.Core.Emulator.OperatingSystem.Structures;
+using Spice86.Shared.Interfaces;
+using Spice86.Shared.Utils;
+
+/// <inheritdoc cref="IDosPathResolver" />
+public class DosPathResolver : IDosPathResolver {
+
+    /// <inheritdoc />
+    public IDictionary<char, MountedFolder> DriveMap { get; private set; } = new Dictionary<char, MountedFolder>();
+
+    /// <inheritdoc />
+    public char CurrentDrive { get; private set; }
+
+    /// <inheritdoc />
+    public string CurrentHostDirectory => ConvertUtils.ToSlashPath(DriveMap[CurrentDrive].FullName);
+
+    /// <inheritdoc />
+    public DosFileOperationResult GetCurrentDosDirectory(byte driveNumber, out string currentDir) {
+        //default drives
+        if (driveNumber == 0 && DriveMap.Any()) {
+            MountedFolder mountedFolder = DriveMap[CurrentDrive];
+            currentDir = Path.GetRelativePath(mountedFolder.MountPoint, mountedFolder.CurrentDirectory).ToUpperInvariant();
+            return DosFileOperationResult.NoValue();
+        }
+        else if (DriveMap.TryGetValue(DriveLetters[driveNumber - 1], out MountedFolder? mountedFolder)) {
+            currentDir = Path.GetRelativePath(mountedFolder.MountPoint, mountedFolder.CurrentDirectory).ToUpperInvariant();
+            return DosFileOperationResult.NoValue();
+        }
+        currentDir = "";
+        return DosFileOperationResult.Error(ErrorCode.InvalidDrive);
+    }
+
+
+    private readonly ILoggerService _loggerService;
+
+    /// <summary>
+    /// Initializes a new instance.
+    /// </summary>
+    /// <param name="loggerService">The logger service implementation.</param>
+    /// <param name="configuration">The emulator configuration.</param>
+    public DosPathResolver(ILoggerService loggerService, Configuration configuration) {
+        _loggerService = loggerService;
+        DriveMap = InitializeDriveMap(configuration);
+        CurrentDrive = 'C';
+        SetCurrentDir(@"C:\");
+    }
+
+    private static string GetExeParentFolder(Configuration configuration) {
```

Comment:
Why not pass in a `string?` and call the method `GetParentFolder`?

Reply:
Done.